core/services/gateway/handlers/functions/orm.go

```diff
@@ -0,0 +1,130 @@
+package functions
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/functions/generated/functions_router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+//go:generate mockery --quiet --name ORM --output ./mocks/ --case=underscore
+
+type ORM interface {
+	GetSubscriptions(offset, limit uint, qopts ...pg.QOpt) ([]CachedSubscription, error)
+	UpsertSubscription(subscription CachedSubscription, qopts ...pg.QOpt) error
+}
+
+type orm struct {
+	q                     pg.Q
+	routerContractAddress common.Address
+}
+
+var _ ORM = (*orm)(nil)
+var (
+	ErrInvalidParameters = errors.New("invalid parameters provided to create a subscription cache ORM")
+)
+
+const (
+	tableName = "functions_subscriptions"
+)
+
+type cachedSubscriptionRow struct {
+	SubscriptionID        uint64
+	Owner                 common.Address
+	Balance               int64
+	BlockedBalance        int64
+	ProposedOwner         common.Address
+	Consumers             pq.ByteaArray
+	Flags                 []uint8
+	RouterContractAddress common.Address
+}
+
+func NewORM(db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig, routerContractAddress common.Address) (ORM, error) {
+	if db == nil || cfg == nil || lggr == nil || routerContractAddress == *new(common.Address) {
```

Comment:
nit: you can also do "... == common.Address{}"

Reply:
😄 I did not like the `*new(common.Address)` either, I was doing ` == common.Address{}` before, but was getting a syntax error. Just needed to wrap it within parenthesis `(common.Address{})`. 👍🏼  changed